include/distr_queue.h

```diff
@@ -51,28 +51,19 @@ class distr_queue {
 	 * Submits a command group to the queue.
 	 *
 	 * Invoke via `q.submit(celerity::allow_by_ref, [&](celerity::handler &cgh) {...})`.
-	 *
-	 * With this overload, CGF may capture by-reference. This may lead to lifetime issues with asynchronous execution, so using the `submit(cgf)` overload is
-	 * preferred in the common case.
 	 */
 	template <typename CGF>
-	void submit(allow_by_ref_t, CGF cgf) { // NOLINT(readability-convert-member-functions-to-static)
-		// (Note while this function could be made static, it must not be! Otherwise we can't be sure the runtime has been initialized.)
-		detail::runtime::get_instance().get_task_manager().submit_command_group(std::move(cgf));
+	[[deprecated("This overload is no longer required to capture by reference")]] void submit(allow_by_ref_t /* tag */, CGF cgf) {
```

Comment:
Interestingly Clang only seems to emit one deprecation warning for this line (if the function is deprecated it will print that warning, otherwise the one for `allow_by_ref_t`), so we are not unnecessarily spamming the user. Given this I would say it's preferable to also deprecate the type, because it actually *is* deprecated. We could only deprecate the type, but then the deprecation won't show up in IDEs (e.g. VSCode prints it with strikethrough).

Reply:
Okay! If this does not produce false-positives warnings in this header, I of course agree - both are actually deprecated.